aten/src/ATen/native/cuda/ScaledGroupMM.cu

```diff
@@ -41,16 +44,87 @@ C10_DIAGNOSTIC_PUSH_AND_IGNORED_IF_DEFINED("-Wunused-but-set-parameter")
 
 #include <cute/atom/mma_atom.hpp>
 #include <cutlass/gemm/dispatch_policy.hpp>
+#include <cutlass/gemm/group_array_problem_shape.hpp>
 #include <cutlass/gemm/kernel/gemm_universal.hpp>
 #include <cutlass/util/packed_stride.hpp>
 
+// Added for SM100 support
+#include <cutlass/epilogue/collective/default_epilogue.hpp>
+#include <cutlass/epilogue/thread/activation.h>
+#include <cutlass/epilogue/thread/linear_combination.h>
+#include <cutlass/gemm/kernel/tile_scheduler_params.h>
+#include <cutlass/tensor_ref.h>
+
+#if (defined(CUDA_VERSION) && CUDA_VERSION >= 12080) && \
+    (defined(CUTLASS_ARCH_MMA_SM100A_SUPPORTED) ||      \
+     defined(CUTLASS_ARCH_MMA_SM100_SUPPORTED))
+#define BUILD_SM100_KERNEL
+#endif
+
+C10_DIAGNOSTIC_POP()
 C10_DIAGNOSTIC_POP()
 C10_DIAGNOSTIC_POP()
 
+namespace at::cuda::detail {
+
+GroupCountInfo get_group_count(
```

Comment:
I noticed that the 2D/3D logic is used across multiple functions, so I consolidated it into a struct for better reuse and readability.